### PR TITLE
improvement/better parsing

### PR DESCRIPTION
**Pull request by TorHou**
Fasta files don't necessarily have the whole sequence in one line for each id.

### CODE REVIEW

File: scripts/split_identify_fasta.py [hunk 1]
```diff
@@ -11,35 +11,44 @@
 parser.add_argument("--Ns", type = int, default = 5000, help = "Number of Ns to split by [5000]")
 
 args = parser.parse_args()
+def read_fasta(fp):
```
Comment (evanbiederstedt) [inline scripts/split_identify_fasta.py:15]: You might want to leave a comment as to how you're parsing this fasta. It's arguably overkill, but maybe just a comment so others immediately see what the function does, and how you're parsing the data. Nothing too crazy.

My philosophy is, whatever you could do now which stops confused github issues in the future, please do it :) You can imagine not everyone using this script is 100% familiar with python syntax.

File: scripts/split_identify_fasta.py [hunk 2]
```diff
@@ -11,35 +11,44 @@
...
-                    outf.write(">" + args.prefix + "_" + rid + "_" + str(partnr) + "\n")
-                    outf.write(part + "\n")
-
-                    partnr += 1
-                    start = nstop
-                part = s[start:]
-                outf.write(">" + args.prefix + "_" + rid + "_" + str(partnr) + "\n")
+        for fid, seqt in read_fasta(f):
+            seq = seqt.upper()
+            if seq[0] not in {'A', 'G', 'C', 'T', 'N'}:
```
Comment (evanbiederstedt) [inline scripts/split_identify_fasta.py]: This is true for human assemblies---AGCTN. 

The issue came up for plants and viruses that these FASTAs commonly contain e.g. R, Y, K, M. There's also U. 

A more informative check might be useful---there are other ways to check that the FASTA is the correct format.

File: scripts/split_identify_fasta.py [hunk 3]
```diff
@@ -11,35 +11,44 @@
...
+                print("Error found in input file. Make sure the input is in correct fasta format.")
+                sys.exit()
+            print(fid + "... ", end = "")
+
+            s = seq.rstrip().upper().lstrip("N").rstrip("N")
+            matches = re.finditer("N{" + str(args.Ns) + ",}", s)
+            start = 0
```
Comment (evanbiederstedt) [inline scripts/split_identify_fasta.py:39]: I always worry when I see this in python scripts, as it reminds me of C code----there are normally quicker ways algorithmically to do things besides iterating like this. 

Nonetheless, it works :) You might want to investigate whether this part could be more efficient---I don't have strong ideas at the moment.

**Review by evanbiederstedt — APPROVED** [review]
A few modifications requested/questions, but it LGTM

File: scripts/split_identify_fasta.py (hunk 3 above)
Comment (TorHou) [reply scripts/split_identify_fasta.py:39]: It's not something that will be used very often. I am certain there are faster ways to do this, but the bottleneck of the pipeline will definitely be elsewhere.

File: scripts/split_identify_fasta.py (hunk 1 above)
Comment (TorHou) [reply scripts/split_identify_fasta.py:15]: It's not super easy to explain this. But I added a comment that points the interested person in the right direction.

File: scripts/split_identify_fasta.py (hunk 2 above)
Comment (TorHou) [reply scripts/split_identify_fasta.py]: I added more letters. This is just a quick check however, I don't want to do a comprehensive check if this is proper fasta. Somebody using the pipeline should have some prior knowledge of how fasta looks like.

Comment (evanbiederstedt) [reply scripts/split_identify_fasta.py]: > Somebody using the pipeline should have some prior knowledge of how fasta looks like.

I agree!

I could see this as an annoying snag though, which drives bioinformaticians crazy....these edge cases which you spend your days trying to figure out :)

File: scripts/split_identify_fasta.py (hunk 1 above)
Comment (evanbiederstedt) [reply scripts/split_identify_fasta.py:15]: > It's not super easy to explain this.

You don't need to necessarily over explain. 

I think commenting what the basic python does in the function works too (for Alex to follow 🤫) ;)

File: scripts/split_identify_fasta.py (hunk 3 above)
Comment (evanbiederstedt) [reply scripts/split_identify_fasta.py:39]: fair enough

**Comment by evanbiederstedt** [comment]
Feel free to merge, if ready @TorHou 

I'm satisfied :)

**Pull request merged**